### DISCUSSION TITLE
[Question] Is there any technical advantage from the architecture amd64 rename?

### DISCUSSION THREAD

**Discussion by lat0z** [terramate-io/terramate#1644] [Q&A] [answered]
From your releaser file,  amd64 architecture is renamed to x86_64.
It would be easy to use your tool if it follows the release "standard" from Terraform which is compliant with the use of the Docker native args TARGETARCH and TARGETOS, to build multi-architecture container images. 
Example 

`wget -O terraform.zip https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_${TARGETOS}_${TARGETARCH}.zip`

**Comment by soerenmartius** [comment]
Hey @lat0z,

Thanks for reaching out. I forwarded your request to our team. @mariux or @i4ki will provide feedback shortly.

**Comment by i4ki** [comment, marked as answer]
Hey @lat0z 

The distribution of `x86_64` makes it easy to download and install on Unix systems by just doing:

```
version="0.3.0"
wget https://github.com/terramate-io/terramate/releases/download/v${version}/terramate_${version}_linux_$(uname -m).tar.gz
```
But this is not a huge advantage because some archs returned by `uname` are not distributed by Terramate (eg.: i686, etc).

We discussed the topic and decided to distribute both arch names. We cannot just change the arch names now because too many packaging tools already rely on the current arch names, also many CI/CDs out there installing from the release assets would just break if we change this. 

Then we are going to release both arch names but we still need to figure out the best way to do it when using [goreleaser](https://goreleaser.com/).
It looks like goreleaser does not support this case but I asked a question in their community forum: https://github.com/orgs/goreleaser/discussions/4171

Alternatively, we can do some hacks, for example, having a post-release script that copy/rename the `x86_64` and `i386` archives and update the GitHub release with the `gh release upload` (see [here](https://cli.github.com/manual/gh_release_upload)).

Thanks for bringing this to our attention.
